persistence/src/main/java/edu/unc/lib/dl/services/DigitalObjectManagerImpl.java

```diff
@@ -623,29 +625,42 @@ public String getLastModified() {
 	 * @throws FedoraException
 	 */
 	private DatastreamDocument getXMLDatastreamIfExists(PID pid, String datastreamName) throws FedoraException {
-
-		if (tripleStoreQueryService.hasDisseminator(pid, datastreamName)) {
-			MIMETypedStream mts = accessClient.getDatastreamDissemination(pid, datastreamName, null);
-
-			// Capture the last modified date of the datastream to return with the document
-			// Fri, 06 Feb 2015 14:54:45 GMT
-			String lastModified = null;
-			for (edu.unc.lib.dl.fedora.types.Property header : mts.getHeader().getProperty()) {
-				if ("Last-Modified".equals(header.getName())) {
-					lastModified = header.getValue();
-					break;
+		
+		log.debug("Attempting to get datastream " + datastreamName + " for object " + pid);
+		
+		try {
+			
+			while (true) {
+		
+				edu.unc.lib.dl.fedora.types.Datastream datastream = managementClient.getDatastream(pid, datastreamName);
+			
+				if (datastream == null) {
+					return null;
 				}
+				
+				log.debug("Got datastream, attempting to get dissemination version with create date " + managementClient.getDatastream(pid, datastreamName).getCreateDate());
```

Comment:
Guessing this is just in here for testing?  Probably don't want it re-retrieval the datastream profile.